src/dispatch/conversation/flows.py

```diff
@@ -146,53 +146,69 @@ def create_incident_conversation(incident: Incident, db_session: SessionLocal):
     return incident.conversation
 
 
-def archive_conversation(incident: Incident, db_session: SessionLocal):
+def archive_conversation(subject: Subject, db_session: SessionLocal):
     """Archives a conversation."""
-    if not incident.conversation:
-        log.warning("Conversation not archived. No conversation available for this incident.")
+    if not subject.conversation:
+        log.warning("Conversation not archived. No conversation available for this subject.")
         return
 
     plugin = plugin_service.get_active_instance(
-        db_session=db_session, project_id=incident.project.id, plugin_type="conversation"
+        db_session=db_session, project_id=subject.project.id, plugin_type="conversation"
     )
     if not plugin:
         log.warning("Conversation not archived. No conversation plugin enabled.")
         return
 
     try:
-        plugin.instance.archive(incident.conversation.channel_id)
+        plugin.instance.archive(subject.conversation.channel_id)
```

Comment:
Mildly worried that we'd be able to call `archive` on Channels that stored thread based cases 😳

Reply:
The only the caller to `archive_conversation` from case/flows.py checks `dedicated_channel` first:
`    if case.dedicated_channel:`
`        conversation_flows.archive_conversation(subject=case, db_session=db_session)`